src/sql/helpers.js

```diff
@@ -10,32 +10,32 @@ const jsonbObjectAgg = (aggregationColumnKey, nestedFields, aggregationJsonKey,
   const jsonBuildObjectSql = `json_build_object(${jsonObjectProps.join(', ')})`;
 
   // Set column {key} as the key for each value
-  const jsonbObjectAggSql = `jsonb_object_agg(${aggregationColumnKey}, ${jsonBuildObjectSql})`;
+  const jsonbObjectAggSql = `jsonb_object_agg(${collapseKey}, ${jsonBuildObjectSql})`;
 
   // When there is no row to aggregate, json_object_agg throws an error
   // So we need to handle this case outside jsonb_object_agg with coalesce
   // Solution based on https://stackoverflow.com/a/33305456
   const handleNoExecutionsFoundSql = (
     `COALESCE(${jsonbObjectAggSql}
       FILTER(
-        WHERE ${aggregationColumnKey} IS NOT NULL
+        WHERE ${collapseKey} IS NOT NULL
       ),
       '{}'::jsonb
     )`
   );
 
-  return sql.raw(`${handleNoExecutionsFoundSql} as ${aggregationJsonKey}`);
+  return sql.raw(`${handleNoExecutionsFoundSql} as ${collapseKeyNewName}`);
 };
 
 /**
  *
  * @param {*} originalQuery Query that fetches the sql data
  * @param {*} rootFields Fields that shouldn't be aggregated
- * @param {*} nestedFields Fields that vary across each aggregationColumnName,
+ * @param {*} fieldsToCollapse Fields that vary across each aggregationColumnName,
  * this is what we want to aggregate into one object for each aggregationColumnName
- * @param {*} aggregationColumnKey The column by which we want to perform the aggregation,
+ * @param {*} collapseKey The column by which we want to perform the aggregation,
  * the object will contain one key for each aggregationColumnName value
- * @param {*} aggregationJsonKey The key that we want to be added holding all the aggregated data
+ * @param {*} collapseKeyNewName The key that we want to be added holding all the aggregated data
```

Comment:
These `collapseKeysIntoObject` and `collapseKeyIntoArray` methods seem really complex and are hard to read. I can't come up with a way to make them easier to read, I think that the problem is the generalisation. Do we really need to have "magic" methods like this that can handle all sorts of fields? Does the benefit of reducing code outweigh the addition of extra complexity?